language/go/generate.go

```diff
@@ -517,7 +519,16 @@ func (g *generator) generateTest(pkg *goPackage, library string) *rule.Rule {
 	if !pkg.test.sources.hasGo() {
 		return goTest // empty
 	}
-	g.setCommonAttrs(goTest, pkg.rel, nil, pkg.test, library)
+	var embed string
+	for src := range pkg.test.sources.strs {
+		f, err := parser.ParseFile(token.NewFileSet(), filepath.Join(pkg.dir, src), nil, parser.PackageClauseOnly)
```

Comment:
Let's not re-open and re-parse these files. Instead, it would be better to set a `hasInternalTest bool` flag in the `goTarget` for the test.